### ISSUE TITLE
Extension overrides fontStyle setting for comments

### ISSUE THREAD

**Issue by d108** [swift-server/vscode-swift#543]
**Describe the bug**

VSCode-Swift overrides fontStyle setting for comment styling.

**To Reproduce**

1. Go to settings.json and add the following:
```
"editor.tokenColorCustomizations": 
{
    "comments": 
    { 
        "fontStyle": "bold"
    }
}
```
2. Open a .swift file.
3. See that comments are not bold.
4. Start typing a comment and see that the font style is overridden.

**Expected behavior**

Comments will display with bold font style when the above customization is added to settings.json.

**Environment**

 - OS: macOS 12+
 - Swift version: 5.7
 - Visual Studio Code version: 1.78
 - vscode-swift version: 1.3

**Comment by adam-fowler** [comment]
The extension doesn't do anything with `tokenColorCustomizations`. There is a chance sourcekit-lsp isn't identifying comments correctly for styling. Do comments get bold styling with the extension disabled?

**Comment by adam-fowler** [comment]
I cannot replicate this issue. 
- What theme are you using?
- Don't want to ask a stupid question but you did save the settings.json?

**Comment by d108** [comment]
Thank you for investigating the matter. Below are my responses to your inquiries.

> Do comments get bold styling with the extension disabled?

Yes, but the bold styling is removed immediately after enabling the extension. In fact, it goes to italics.

> Also does this work with other language extensions?

Yes. For example, Python and C/C++.

> Also do you have semantic highlighting on? If so you might need to edit editor.semanticTokenColorCustomizations instead.

Yes. However, toggling the setting `editor.semanticTokenColorCustomizations` enables bold styling.

> What theme are you using?

I am currently using PaperColor Redux.

> Don't want to ask a stupid question but you did save the settings.json?

I believe so. I have tried restarting VS Code several times.

In conclusion, the extension will allow bold styling for comments when `editor.semanticTokenColorCustomizations` is set to `false.` However, the issue persists when `editor.semanticTokenColorCustomizations` is set to `true.`

Based on your suggestion that the problem might be related to sourcekit-lsp, I have confirmed that the issue persists even with Swift 5.8. Is this the appropriate way to test the impact of sourcekit-lsp?

Disabling `semanticTokenColorCustomizations` might work well enough for me to close the issue. I'm not sure if it's a bug or not at this point. What do you think?

**Comment by adam-fowler** [comment]
So I installed papercolor redux and still cannot replicate this. 
I did notice though that the papercolor redux theme does use semantic customisations for comments if an extension provides semanticTokens. So maybe you need to override those. I don't know what takes precedence. From what I can tell the C/C++ extension does not provide semanticTokens.

So yeah if you have semantic token highlighting enabled you can set comments to bold using
```jsonc
    "editor.semanticTokenColorCustomizations": {
        "rules": {
            "comment": {
                "bold": true
            }
        }
    }
```

**Comment by d108** [comment]
Thank you for testing out the PaperColor Redux theme. I noticed that I got the same results when I switched between themes.  However, knowing the potential impact of semantic customizations is beneficial.

From our discussion, I understand the problem much better now. While adding the above change to `editor.semanticTokenColorCustomizations` > `rules` > `comment` > `bold: true` did not cure my problem, I narrowed it down to a conflicting configuration setting. I've shared the ["editor.tokenColorCustomizations"](https://gist.github.com/d108/0a612c6b71eda56fa56c1ec732c5d13e) dictionary from my settings.json. I've posted a compressed version of the relevant portion below.

```js
"textMateRules": [
    {
        "name": "One Dark italic",
        "scope": [
            "comment",
        "settings": {
            "fontStyle": "italic"
        }
    },
```

Under the key `textMateRules,` the culprit for italic styling appears to come from a rule defined there for the One Dark Italic Theme I no longer have installed. Strangely, the italics styling only showed up when using vscode-swift. I did not initially suspect those configuration settings because I had never touched them.

I appreciate your assistance in troubleshooting the issue. The problem was related to my configuration, which made it difficult to reproduce. Thanks to your guidance, I am better equipped to troubleshoot similar issues in the future. 

I was able to resolve the problem by removing the conflicting theme leftovers. As a result, I am now able to view bold styling in comments with `"editor.semanticHighlighting.enabled"` set to `true.`

If you're interested, you can try reproducing the issue yourself. Thank you again for your help.
